torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1579,6 +1646,16 @@ def _verify(expr, potential_expr):
         # these should probably get reported in tests too
         if not _simplified:
             for symbol, sources in symbol_to_source.items():
+
+                # TODO(voz): Should this valid range actually come from the passed in dynamic_ranges?
+                # On one hand, yes! It's a lot more like the contract we agreed upon with verification
+                # with regards to produce_guards in the first place.
+                # On the other hand, self.var_to_range is pretty useful to check too!
```

Comment:
@ezyang opine here please! :)

Reply:
Maybe this is more convenient, but you're mixing up two concerns here.

The user defined range doesn't have to be consistent with the size hint. However, if this happens, it can be tested IMMEDIATELY, whenever you get the user range, you don't have to wait until the end. 

We can also refine the range, e.g., by work done in https://github.com/pytorch/pytorch/pull/96616 But if we refine a range out of what the actual sizevar is, that's an INTERNAL error from the framework.

We want to give different error messages in these two cases. So I prefer not conflating.